### ISSUE TITLE
why D_real.backward(one) and D_fake.backward(mone)?

### ISSUE THREAD

**Issue by geekfeiw** [caogang/wgan-gp#43]
Thanks for your excellent work. I have two confusion on the operation said on the Title.

1:  what are one/mone doing here

2: I read some GAN codes, they first compute loss, then to do backward （loss_D_real.backward() or loss_D_fake.backward() ）.  What's the reason you not using loss?

**Comment by micklexqg** [comment]
same confusion. have you figured out?
why the computed loss Wasserstein_D and D_cost is not backward but only for plot?

**Comment by HymEric** [comment]
I gauss I figured it out. one=Tesor[1] mnoe=Tensor=[-1] During the Discriminator training, we want to maximize the loss for real data, so we use backward(mnoe) because the default loss minimizes the loss so we should use -1 as the argument. We want to minimize the loss of fake data, so we use backward(one). And the same with Generator training.

Please refer to the argument influence of [backward()](https://pytorch.org/tutorials/beginner/blitz/autograd_tutorial.html#sphx-glr-beginner-blitz-autograd-tutorial-py)
